src/main/java/com/spotify/docker/client/messages/swarm/DnsConfig.java

```diff
@@ -0,0 +1,152 @@
+import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import javax.annotation.Nullable;
+
+@AutoValue
+@JsonAutoDetect(fieldVisibility = ANY, getterVisibility = NONE, setterVisibility = NONE)
+public abstract class DnsConfig {
+
+  @Nullable
+  @JsonProperty("Nameservers")
+  public abstract ImmutableList<String> nameServers();
+
+  @Nullable
+  @JsonProperty("Search")
+  public abstract ImmutableList<String> search();
+
+  @Nullable
+  @JsonProperty("Options")
+  public abstract ImmutableList<String> options();
+
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    public abstract Builder nameServers(String... nameServers);
+
+    public abstract Builder nameServers(List<String> nameServers);
+
+    /**
+    * @deprecated  As of release 7.0.0, replaced by {@link #nameServers(String...)}.
+    */
+    @Deprecated
+    public Builder withNameServers(final String... nameServers) {
+      if (nameServers != null && nameServers.length > 0) {
+        nameServers(nameServers);
+      }
+      return this;
+    }
+
+    /**
+    * @deprecated  As of release 7.0.0, replaced by {@link #nameServers(List)}.
+    */
+    @Deprecated
+    public Builder withNameServers(final List<String> nameServers) {
+      nameServers(nameServers);
+      return this;
+    }
+    
+    public abstract Builder search(String... search);
+
+    public abstract Builder search(List<String> search);
+
+    /**
+    * @deprecated  As of release 7.0.0, replaced by {@link #search(String...)}.
+    */
+    @Deprecated
+    public Builder withsearch(final String... search) {
+      if (search != null && search.length > 0) {
+        search(search);
+      }
+      return this;
+    }
+
+    /**
+    * @deprecated  As of release 7.0.0, replaced by {@link #search(List)}.
+    */
+    @Deprecated
+    public Builder withsearch(final List<String> search) {
+      search(search);
+      return this;
+    }
+    
+    public abstract Builder options(String... options);
+
+    public abstract Builder options(List<String> options);
+
+    /**
+    * @deprecated  As of release 7.0.0, replaced by {@link #options(String...)}.
+    */
+    @Deprecated
+    public Builder withoptions(final String... options) {
+      if (options != null && options.length > 0) {
+        options(options);
+      }
+      return this;
+    }
+
+    /**
+    * @deprecated  As of release 7.0.0, replaced by {@link #options(List)}.
+    */
+    @Deprecated
+    public Builder withoptions(final List<String> options) {
+      options(options);
+      return this;
+    }
+
+    public abstract com.spotify.docker.client.messages.swarm.DnsConfig build();
```

Comment:
You should not need to give the full package of the class here. Just the class name will do. 

(Your IDE may have inserted this for you. I know mine did that for a while, but I don't remember how I was able to get it to stop doing that.)